features/steps/steps.py

```diff
@@ -66,6 +68,51 @@ def fmt(x):
             return "...".join((v[:25], v[-7:]))
         return v
 
+def is_closed(context, instance):
+    entity_contexts = recurrently_get_entity_attr(context, instance, 'IfcRepresentation', 'ContextOfItems', set())
+    precision = get_precision_from_contexts(entity_contexts)
+    points_coordinates = get_points(instance)
+    return math.dist(points_coordinates[0], points_coordinates[-1]) < precision
+
+def get_points(inst, return_type='coord'):
+    if inst.is_a().startswith('IfcCartesianPointList'):
+        return inst.CoordList
+    elif inst.is_a('IfcPolyline'):
+        if return_type == 'coord':
+            return [p.Coordinates for p in inst.Points]
+        elif return_type == 'points':
+            return inst.Points
+    elif inst.is_a('IfcPolyLoop'):
+        if return_type == 'coord':
+            return [p.Coordinates for p in inst.Polygon]
+        elif return_type == 'points':
+            return inst.Polygon
+    else:
+        raise NotImplementedError(f'get_points() not implemented on {inst.is_a}')
+
+def recurrently_get_entity_attr(ifc_context, ent, entity_to_look_for, attr_to_get, attr_found):
```

Comment:
I think this looks good:

Just two small comments:

- I really prefer `inst` instead of `ent` as the variable name. To prevent confusion between entity names and instances.
- Good job on not doing `attr_found=set()` because default values are only initialized once so it would get all occurences to fill into the same set occurence. But I would prefer:

```python
def recurrently_get_entity_attr(..., attr_found=None):
  if attr_found is None:
    attr_found=set()
  ...
```